pkg/asset/manifests/topologies.go

```diff
@@ -10,17 +10,18 @@ import (
 // determineTopologies determines the Infrastructure CR's
 // infrastructureTopology and controlPlaneTopology given an install config file
 func determineTopologies(installConfig *types.InstallConfig) (controlPlaneTopology configv1.TopologyMode, infrastructureTopology configv1.TopologyMode) {
-	controlPlaneTopology = configv1.HighlyAvailableTopologyMode
-
 	controlPlaneReplicas := ptr.Deref(installConfig.ControlPlane.Replicas, 3)
-	if controlPlaneReplicas == 2 {
+	switch controlPlaneReplicas {
+	case 1:
+		controlPlaneTopology = configv1.SingleReplicaTopologyMode
+	case 2:
 		controlPlaneTopology = configv1.DualReplicaTopologyMode
+	default:
+		controlPlaneTopology = configv1.HighlyAvailableTopologyMode
+	}
 
-		if installConfig.Arbiter != nil && ptr.Deref(installConfig.Arbiter.Replicas, 0) != 0 {
-			controlPlaneTopology = configv1.HighlyAvailableArbiterMode
-		}
-	} else if controlPlaneReplicas < 3 {
-		controlPlaneTopology = configv1.SingleReplicaTopologyMode
+	if controlPlaneReplicas >= 2 && installConfig.Arbiter != nil && ptr.Deref(installConfig.Arbiter.Replicas, 0) != 0 {
```

Comment:
This may be overly pedantic, but would our validation catch negative arbiters?

Reply:
Good call out, right now we do have a check in the validation step, https://github.com/openshift/installer/blob/main/pkg/asset/agent/installconfig.go#L217-L232 that would catch negative numbers